### ISSUE TITLE
Formatted (HTML) messages with markdown features are rendered incorrectly as markdown

### ISSUE THREAD

**Issue by tulir** [vector-im/element-android#350]
When the `formatted_body` field of a `"format": "org.matrix.custom.html"` message contains special markdown characters, they're rendered as markdown instead of as plain text. This could be an actual problem if you escape markdown characters in a message that contains other formatting

<details>
<summary>Exaggerated example</summary>

### Expected rendering
![riot web](https://user-images.githubusercontent.com/4224639/61151780-64fdbb80-a4ef-11e9-90f2-50583bcaa874.png)
![riot android](https://user-images.githubusercontent.com/4224639/61151845-8ced1f00-a4ef-11e9-9216-ac15e4bac826.png)

### Actual rendering
![riot x](https://user-images.githubusercontent.com/4224639/61151872-99717780-a4ef-11e9-8345-1a7567961616.png)

### Event source
```json
{
  "type": "m.room.message",
  "sender": "@tulir:maunium.net",
  "content": {
    "msgtype": "m.text",
    "body": "Blank message (or is it? 🤔)",
    "format": "org.matrix.custom.html",
    "formatted_body": "---\n> * # [`RiotX` renders this message ***completely incorrectly***](https://github.com/vector-im/riotX-android/issues/289#issuecomment-510894293)\n---"
  },
  "event_id": "$156295723912295JSonL:maunium.net",
  "origin_server_ts": 1562957239380,
  "unsigned": {
    "age": 52
  },
  "room_id": "!tTIucwUZLRtKnXeurb:matrix.org"
}
```
</details>

<details>
<summary>Non-exaggerated example</summary>

### Expected rendering
![riot web](https://user-images.githubusercontent.com/4224639/61152222-7abfb080-a4f0-11e9-87d9-fa0d30c7994b.png)
![riot android](https://user-images.githubusercontent.com/4224639/61152269-9f1b8d00-a4f0-11e9-9d15-9664b5010a04.png)

### Actual rendering
![riot x](https://user-images.githubusercontent.com/4224639/61152284-a9d62200-a4f0-11e9-8f01-2fdf9a11753b.png)

### Event source
```json
{
  "type": "m.room.message",
  "sender": "@tulir:maunium.net",
  "content": {
    "msgtype": "m.text",
    "format": "org.matrix.custom.html",
    "body": "*less exaggerated example*: \\*\\*stars!**",
    "formatted_body": "<em>less exaggerated example</em>: **stars!**"
  },
  "event_id": "$156295796112315whymM:maunium.net",
  "origin_server_ts": 1562957961390,
  "unsigned": {
    "age": 170
  },
  "room_id": "!jhpZBTbckszblMYjMK:matrix.org"
}
```
</details>

<details>
<summary>Real-life example</summary>

[matrix:room/riotx:matrix.org/event/$15630377604255dZoDF:hacklab.fi](https://matrix.to/#/!tTIucwUZLRtKnXeurb:matrix.org/$15630377604255dZoDF:hacklab.fi?via=matrix.org&via=privacytools.io&via=feneas.org)

This one was a reply sent from RiotX, which had indentation in the fallback format for some reason. The indentation of course turned into a code block because that's what a markdown parser does.

This specific example could be fixed by
* fixing whatever caused the indentation, or
* adding reply rendering to RiotX,

but of course neither of those is a proper fix to this issue.

### Expected rendering
![riot web](https://user-images.githubusercontent.com/4224639/61174713-9f795e00-a5ac-11e9-8127-deec6ac0d0d6.png)
![riot android](https://user-images.githubusercontent.com/4224639/61174725-cfc0fc80-a5ac-11e9-818e-6dfb3f3b0402.png)

### Actual rendering
![riot x](https://user-images.githubusercontent.com/4224639/61174722-c899ee80-a5ac-11e9-90b8-054d5b3d5229.png)

### Event source
```json
{
  "content": {
    "body": "><@marmulak:converser.eu> Does the \"black\" theme work for reducing power consumption on oled displays?\n\nYes, the dark theme will reduce power consumption and the black theme will reduce it a bit more",
    "format": "org.matrix.custom.html",
    "formatted_body": "<mx-reply>\n               <blockquote>\n                   <a href=\"https://matrix.to/#/!tTIucwUZLRtKnXeurb:matrix.org/$1563036634937vaAEg:converser.eu\">In reply to</a>\n                   <a href=\"https://matrix.to/#/@marmulak:converser.eu\">@marmulak:converser.eu</a>\n                   <br />\n                   <p>Does the &quot;black&quot; theme work for reducing power consumption on oled displays?</p>\n\n               </blockquote>\n           </mx-reply>\n           Yes, the dark theme will reduce power consumption and the black theme will reduce it a bit more",
    "m.relates_to": {
      "m.in_reply_to": {
        "event_id": "$1563036634937vaAEg:converser.eu"
      }
    },
    "msgtype": "m.text"
  },
  "event_id": "$15630377604255dZoDF:hacklab.fi",
  "origin_server_ts": 1563037760897,
  "sender": "@vurpo:hacklab.fi",
  "type": "m.room.message",
  "unsigned": {
    "age": 630
  },
  "room_id": "!tTIucwUZLRtKnXeurb:matrix.org"
}
```
</details>

**Comment by bmarty** [comment]
I reopen the issue, we will close it once when we can confirm that we do not reproduce it

**Comment by jwh** [comment]
\/\/\/\/

**Comment by cloudrac3r** [comment]
> I reopen the issue, we will close it once when we can confirm that we do not reproduce it

This really is fixed by #6357 and won't be reproduced anymore.

**Comment by bmarty** [comment]
@cloudrac3r OK, thanks!
#6357 is included in Element Android 1.4.26.
